libraries/ms-common/src/stm32f0xx/soft_timer.c

```diff
@@ -203,7 +212,29 @@ static void prv_update_timer(void) {
   // and reenable compares. In the case where there aren't any timers registered, the compare
   // channel is disabled until a new timer is added.
   if (s_timers.head != NULL) {
-    TIM_SetCompare1(TIM2, s_timers.head->expiry_us);
+    uint32_t curr_time = TIM_GetCounter(TIM2);
```

Comment:
This is equivalent to

```
bool min_interval_rollover = min_interval_time < curr_time;
bool next_expiry_rollover = s_timers.head->expiry_rollover_count;
if (min_interval_rollover == next_expiry_rollover) {
  set compare to max(next expiry, min_interval);
} else {
  set compare to (min_interval_rollover > next_expiry_rollover) ? min_interval : next expiry;
}
```

?

Reply:
True, I did it the other way since it was easier to logic out and follow but I've updated it to this.

Reply:
Yeah, yours was definitely more explicit. It might be nice to keep some of the comments explaining how it works/why it's necessary.